cmd/gaia/app/genesis.go

```diff
@@ -42,8 +42,10 @@ type GenesisState struct {
 	GenTxs       []json.RawMessage     `json:"gentxs"`
 }
 
-func NewGenesisState(accounts []GenesisAccount, authData auth.GenesisState, stakeData stake.GenesisState, mintData mint.GenesisState,
-	distrData distr.GenesisState, govData gov.GenesisState, slashingData slashing.GenesisState) GenesisState {
+func NewGenesisState(accounts []GenesisAccount, authData auth.GenesisState,
```

Comment:
These are growing quite large. I think eventually we should just encapsulate all of them into a `GenesisState` structure of some sort and simply pass that structure in.

Reply:
Maybe but I think that type of solution only makes sense when the same group of variables are being passed around between multiple functions, AFAICT this group is only occurring once. Leaving for now

Reply:
Right, it doesn't have to be done in this PR. I'm just stating. The function will be easier to read and call.